src/Symfony/Component/Security/Http/Firewall/SwitchUserListener.php

```diff
@@ -122,7 +122,10 @@ private function attemptSwitchUser(Request $request)
         }
 
         if (false === $this->accessDecisionManager->decide($token, array($this->role))) {
-            throw new AccessDeniedException();
+            $exception = new AccessDeniedException();
+            $exception->setAttributes(array($this->role));
```

Comment:
In your signature it states that you can also add a string and it will internally convert. My suggestion would be to either remove the array usage here or allow only an array item in the signature (my preference).